caikit/interfaces/common/data_model/remote.py

```diff
@@ -146,6 +146,9 @@ class ConnectionInfo(DataObjectBase):
     retries: Optional[int] = 1
     # Runtime specific retry options
     retry_options: Optional[JsonDict] = field(default_factory=dict)
+    # Maximum age for a client channel. Values less then 0 are infinite while 0 means new
+    # channel/session for every request
+    max_session_age: Optional[float] = -1
```

Comment:
Why is this marked `Optional` if the default value is not `None`? I think setting it to `None` would cause the `__post_int__` check to fail.

Reply:
Good call! I updated it to `Union[float, int]` to be explicit about the types checked